src/machine/machine_esp32.go

```diff
@@ -363,38 +363,16 @@ func (spi SPI) Configure(config SPIConfig) error {
 	// (mode), among others.
 	var ctrl2Reg, userReg, pinReg uint32
 	// For mode configuration, see table 29 in the reference manual (page 128).
-	var delayMode uint32
 	switch config.Mode {
 	case 0:
-		delayMode = 2
 	case 1:
-		delayMode = 1
 		userReg |= esp.SPI_USER_CK_OUT_EDGE
 	case 2:
-		delayMode = 1
 		userReg |= esp.SPI_USER_CK_OUT_EDGE
 		pinReg |= esp.SPI_PIN_CK_IDLE_EDGE
 	case 3:
-		delayMode = 2
 		pinReg |= esp.SPI_PIN_CK_IDLE_EDGE
 	}
-	// Extra configuration necessary for correct data input at high frequencies.
-	// This is only necessary when MISO goes through the GPIO matrix (which it
-	// currently does).
-	if config.Frequency >= 40e6 {
-		// Delay mode must be set to 0 and SPI_USR_DUMMY_CYCLELEN should be set
-		// to 0 (the default).
-		userReg |= esp.SPI_USER_USR_DUMMY
-	} else if config.Frequency >= 20e6 {
-		// Nothing to do here, delay mode should be set to 0 according to the
-		// datasheet.
-	} else {
-		// Follow the delay mode as given in table 29 on page 128 of the
-		// reference manual.
-		// Note that this is only specified for SPI frequency of 10MHz and
-		// below (≤Fapb/8), so 13.3MHz appears to be left unspecified.
-		ctrl2Reg |= delayMode << esp.SPI_CTRL2_MOSI_DELAY_MODE_Pos
```

Comment:
I think the mistake is that I did this to the MOSI pin, while it should have been used for MOSI.

Reply:
If I change it to the following, it seems to work fine.

```go
ctrl2Reg |= delayMode << esp.SPI_CTRL2_MISO_DELAY_MODE_Pos
```

![image](https://user-images.githubusercontent.com/9251039/134457812-80d32847-3a36-4efc-8ae2-35cddc1461f1.png)

Reply:
It was at 4MHz that I thought it was working.
At 4Mhz, the read succeeds regardless of the MOSI_DELAY_MODE setting.
However, when MOSI_DELAY_MODE is set at 10 MHz and 13.3 MHz, as shown below, it does not read properly.

![image](https://user-images.githubusercontent.com/9251039/134473099-f6c5a5b4-8ec9-43c9-afcf-8698fa85c5d0.png)

The expected value is 0xE8, but if it doesn't read properly, it will read 0xF4, a value that has been shifted by one bit.

```
MISO_DELAY_MODE
    40   MHz: F4
    20   MHz: F4
    13.3 MHz: F4 !!
    10   MHz: F4 !!
     8   MHz: E8
     6.5 MHz: E8

NO DELAY
    40   MHz: F4
    20   MHz: F4
    13.3 MHz: E8
    10   MHz: E8
     8   MHz: E8
     6.5 MHz: E8
```

It still looks like it is better not to set MOSI_DELAY_MODE.
In the following source, MOSI_DELAY_MODE is not used.

The DELAY_MODE is not used except in the selected slave configuration.

https://github.com/espressif/arduino-esp32/blob/master/tools/sdk/esp32/include/hal/esp32/include/hal/spi_ll.h#L370-L425